### ISSUE TITLE
[BUG] This program was compiled against version 2.6.1 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.21.6).

### ISSUE THREAD

**Issue by ZhouZhaoJi** [android/ndk#1966]
### Description

![image](https://github.com/android/ndk/assets/36101779/aaaf0fb2-b2bb-499e-80ba-8c110d7a8dac)

I have a program that uses grpc-1.49.3. I pushed it to the emulator and ran it, but it outputed this error. My app links grpc libs as static libs, so it should not load a shared lib from OS.

Here are my steps to install the Android SDK.
```
#!/bin/bash

wget https://dl.google.com/android/repository/commandlinetools-linux-10406996_latest.zip
unzip commandlinetools-linux-10406996_latest.zip

mkdir android_sdk
mv cmdline-tools android_sdk
cd android_sdk/cmdline-tools
mkdir latest
mv NOTICE.txt bin lib source.properties latest
cd latest/bin

echo "y" | ./sdkmanager --install "ndk;25.1.8937393" "emulator" "platform-tools" "platforms;android-24" "system-images;android-24;default;arm64-v8a" 
```

Here are my steps to setup the envirnmental variables:
```
#!/bin/bash

export ANDROID_HOME=`pwd`/android_sdk
export PATH=${PATH}:${ANDROID_HOME}/cmdline-tools/latest/bin/
export PATH=${PATH}:${ANDROID_HOME}/emulator
export PATH=${PATH}:${ANDROID_HOME}/platform-tools
```

Here is how I create the avd and run the emulator:
```
echo "no" | avdmanager create avd -n test -k "system-images;android-24;default;arm64-v8a"
emulator -avd test -netdelay none -netspeed full -no-window
```

### Affected versions

Canary

### Canary version

25.1.8937393

### Host OS

Linux

### Host OS version

Ubuntu 22.04

### Affected ABIs

arm64-v8a

### Build system

CMake

### minSdkVersion

24

**Comment by ZhouZhaoJi** [comment]
(Version verification failed in "out/target/product/generic_arm64/gen/STATIC_LIBRARIES/libhwui_static_intermediates/proto/frameworks/base/libs/hwui/protos/hwui.pb.cc".)
hwui.pb.cc is not the part of my program, what is it?

**Comment by DanAlbert** [comment]
libhwui is part of the OS. Without a repro case I've got no idea what to suggest, other than that this doesn't look like an NDK bug. Incorrect linking of grpc seems likely (what you said about static linking ought to be correct, but it's an easy thing to screw up in a complex build). If you haven't used `-Wl,--exclude-libs` to make sure that the grpc symbols are _private_ in your app, it's possible that something in the zygote has already loaded grpc and the system's symbols are taking precedence.

Closing until there's something for us to investigate. If we get a repro case we can do more.
